src/Serializer/AbstractItemNormalizer.php

```diff
@@ -628,6 +628,15 @@ protected function normalizeCollectionOfRelations(ApiProperty $propertyMetadata,
                 throw new UnexpectedValueException('Unexpected non-object element in to-many relation.');
             }
 
+            // update context, if concrete object class deviates from general relation class (e.g. in case of polymorphic resources)
+            $objResourceClass = $this->resourceClassResolver->getResourceClass($obj, $resourceClass);
+            if ($objResourceClass !== $resourceClass) {
+                $context['resource_class'] = $objResourceClass;
+                if ($this->resourceMetadataCollectionFactory) {
+                    $context['operation'] = $this->resourceMetadataCollectionFactory->create($objResourceClass)->getOperation();
```

Comment:
```suggestion
                    $context['operation'] ??= $this->resourceMetadataCollectionFactory?->create($objResourceClass)->getOperation();
```

Would that work to avoid `if`?

Reply:
`$context['operation']` is already populated with the wrong operation in https://github.com/api-platform/core/blob/main/src/Serializer/AbstractItemNormalizer.php#L594
So I actually need to override it here with the correct one.

I could remove if from the earlier location entirely and populate `resource_class` and `operation` only within `normalizeCollectionOfRelations`, then this could work

Reply:
Mhh but why isn't that done in the AbstractItemNormalizer? IMO this behavior should be the same accross all our normalizers no?

Reply:
> Mhh but why isn't that done in the AbstractItemNormalizer?

This is already in `AbstractItemNormalizer`. Not sure I understand.

> Would that work to avoid if?

The first `if` was actually a really bad idea. As this is in a for-loop, context from one item spilled over to other items in case the if-statement was false (was actually the case in the new behat test I introduced). I now changed it, so `resource_class` and `operation` is populated within the for-loop and always overridden.

Do you feel this is a safe approach? Or better to clone the complete `$context` for every object within the collection?

Reply:
We already mutate this object a lot no problem. It's definitely arguable haha.